inference-engine/src/gna_plugin/transformations/convert_dwsc_to_scaleshifts.cpp

```diff
@@ -0,0 +1,246 @@
+        ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{2}, ngraph::Shape{1, 0}));
+
+    auto flat_input_plane = std::make_shared<ngraph::opset7::Reshape>(transposed_input_plane,
+        ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{2},
+            ngraph::Shape{1, shape_size(dwsc->input_value(0).get_shape())}), false);
+
+    // Prepare flat filter data
+    auto filters_const = std::dynamic_pointer_cast<ngraph::Node>(dwsc->get_input_node_shared_ptr(1));
+    auto filters_size = shape_size(filters_const->get_shape());
+
+    auto transposed_filters_const = ngraph::op::util::make_try_fold<ngraph::opset7::Transpose>(filters_const,
+        ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{5}, ngraph::Shape{4, 1, 2, 3, 0}));
+
+    auto flat_filters_plane = ngraph::op::util::make_try_fold<ngraph::opset7::Reshape>(transposed_filters_const,
+        ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{2}, ngraph::Shape{1, filters_size}), false);
+
+    copy_runtime_info(dwsc, {reshaped_input_plane, transposed_input_plane, flat_input_plane, transposed_filters_const, flat_filters_plane});
+
+    // Convert DWSC to a set of diagonal layers
+    auto output_plane = DecomposeDWSC(dwsc, bias_const, fq_bias, flat_input_plane, flat_filters_plane);
+
+    // Restore the original output shape
+    auto result = std::make_shared<ngraph::opset7::Reshape>(output_plane,
+        ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{4},
+            ngraph::Shape{1, output_channel_count, 1, output_width}), false);
+    copy_runtime_info(dwsc, result);
+
+    // We need to put here the original Group Convolution layer name, so the new layer output can be used as a network result
+    std::string result_name = original_last_node->get_friendly_name();
+    replace_node(original_last_node, result);
+    result->set_friendly_name(result_name);
+
+    return true;
+}
+
+ConvertDWSCToScaleShifts::ConvertDWSCToScaleShifts() {
+    MATCHER_SCOPE(ConvertDWSCToScaleShifts);
+
+    auto dwsc = ngraph::pattern::wrap_type<ngraph::opset7::GroupConvolution>(
+        {ngraph::pattern::any_input(), ngraph::pattern::wrap_type<ngraph::opset7::Constant>(ngraph::pattern::rank_equals(5))},
+        ngraph::pattern::rank_equals(4));
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        return Convert(pattern_map.at(dwsc).get_node_shared_ptr(), nullptr, nullptr, nullptr, nullptr);
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(dwsc, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ConvertDWSCBiasToScaleShifts::ConvertDWSCBiasToScaleShifts() {
+    MATCHER_SCOPE(ConvertDWSCBiasToScaleShifts);
+
+    auto dwsc = ngraph::pattern::wrap_type<ngraph::opset7::GroupConvolution>(
```

Comment:
it'd be better to check if GroupConvolution is DWSC right in the matcher subgraph (not in callback), it can be done by passing a custom lambda function in wrap_type